From 93e1b89c07ba37855a3e022ac311268d1ed5728f Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Sat, 6 Aug 2016 21:27:30 -0700
Subject: [PATCH 3/6] add io.h include for portaudio source/sink

The io.h include is conditional when _MSC_VER is defined.
io.h is required under msvc to provide the ::write() call
for the audio overflow and underflow terminal messages.
---
 gr-audio/lib/portaudio/portaudio_sink.cc   | 4 ++++
 gr-audio/lib/portaudio/portaudio_source.cc | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/gr-audio/lib/portaudio/portaudio_sink.cc b/gr-audio/lib/portaudio/portaudio_sink.cc
index 2397b65..e6698ad 100644
--- a/gr-audio/lib/portaudio/portaudio_sink.cc
+++ b/gr-audio/lib/portaudio/portaudio_sink.cc
@@ -24,6 +24,10 @@
 #include "config.h"
 #endif
 
+#ifdef _MSC_VER
+#include <io.h>
+#endif
+
 #include "audio_registry.h"
 #include <portaudio_sink.h>
 #include <portaudio_impl.h>
diff --git a/gr-audio/lib/portaudio/portaudio_source.cc b/gr-audio/lib/portaudio/portaudio_source.cc
index 1e1bbfa..7f2e8f3 100644
--- a/gr-audio/lib/portaudio/portaudio_source.cc
+++ b/gr-audio/lib/portaudio/portaudio_source.cc
@@ -24,6 +24,10 @@
 #include "config.h"
 #endif
 
+#ifdef _MSC_VER
+#include <io.h>
+#endif
+
 #include "audio_registry.h"
 #include <portaudio_source.h>
 #include <portaudio_impl.h>
-- 
2.1.4

